### ISSUE TITLE
JSON-RPC: support for getapplicationlog

### ISSUE THREAD

**Issue by nimmortal** [neow3j/neow3j#16]
Hello.
Could you please add support for getapplicationlog feature.
This is crucial ability to read contract events.

Thanks.

**Comment by gsmachado** [comment]
Hello @nimmortal,
Yes, this is in our pipeline and it will be implemented soon.

**Comment by nimmortal** [comment]
@gsmachado, hello again.

Developing my own contract I reveal, that notification format can be different than simple array. It also may be a map. By the moment such response is not supported by neow3j, but I am planning to add this.

Example:
```json
{
  "jsonrpc": "2.0",
  "id": 1,
  "result": {
    "txid": "0x8e34291755a55090cb7e7cb5d6c5d24412f91b68f52a90a8f74083f0d72f9c3e",
    "executions": [
      {
        "trigger": "Application",
        "contract": "0x0674dd332fc71c0195efcdcf24300788811c847b",
        "vmstate": "HALT, BREAK",
        "gas_consumed": "2.995",
        "stack": [
          {
            "type": "ByteArray",
            "value": ""
          },
          {
            "type": "ByteArray",
            "value": ""
          },
          {
            "type": "Integer",
            "value": "1"
          }
        ],
        "notifications": [
          {
            "contract": "0xef182f4977544adb207507b0c8c6c3ec1749c7df",
            "state": {
              "type": "Map",
              "value": [
                {
                  "key": {
                    "type": "ByteArray",
                    "value": "67656e6573"
                  },
                  "value": {
                    "type": "ByteArray",
                    "value": "7ee96f2f7592e9de2f5dce5e4df72ddf6abe16ede807fe0d8f636847fe5a1998f98a79bf17a90dce5cac48f91c56364d2d2d5810b300"
                  }
                },
                {
                  "key": {
                    "type": "ByteArray",
                    "value": "62697274685f74696d65"
                  },
                  "value": {
                    "type": "Integer",
                    "value": "1557911235"
                  }
                },
                {
                  "key": {
                    "type": "ByteArray",
                    "value": "636f6f6c646f776e5f656e645f74696d65"
                  },
                  "value": {
                    "type": "ByteArray",
                    "value": ""
                  }
                },
                {
                  "key": {
                    "type": "ByteArray",
                    "value": "6d6f6d5f6964"
                  },
                  "value": {
                    "type": "ByteArray",
                    "value": ""
                  }
                },
                {
                  "key": {
                    "type": "ByteArray",
                    "value": "6461645f6964"
                  },
                  "value": {
                    "type": "ByteArray",
                    "value": ""
                  }
                },
                {
                  "key": {
                    "type": "ByteArray",
                    "value": "636f6f6c646f776e5f696e646578"
                  },
                  "value": {
                    "type": "ByteArray",
                    "value": ""
                  }
                },
                {
                  "key": {
                    "type": "ByteArray",
                    "value": "67656e65726174696f6e"
                  },
                  "value": {
                    "type": "ByteArray",
                    "value": ""
                  }
                },
                {
                  "key": {
                    "type": "ByteArray",
                    "value": "6f7074696f6e616c"
                  },
                  "value": {
                    "type": "ByteArray",
                    "value": ""
                  }
                }
              ]
            }
          },
          {
            "contract": "0xef182f4977544adb207507b0c8c6c3ec1749c7df",
            "state": {
              "type": "Array",
              "value": [
                {
                  "type": "ByteArray",
                  "value": "42697274685769746847656e65726174696f6e"
                },
                {
                  "type": "ByteArray",
                  "value": "23ba2703c53263e8d6e522dc32203339dcd8eee9"
                },
                {
                  "type": "Integer",
                  "value": "1"
                },
                {
                  "type": "ByteArray",
                  "value": ""
                },
                {
                  "type": "ByteArray",
                  "value": ""
                },
                {
                  "type": "ByteArray",
                  "value": "7ee96f2f7592e9de2f5dce5e4df72ddf6abe16ede807fe0d8f636847fe5a1998f98a79bf17a90dce5cac48f91c56364d2d2d5810b300"
                },
                {
                  "type": "ByteArray",
                  "value": ""
                },
                {
                  "type": "ByteArray",
                  "value": ""
                },
                {
                  "type": "Integer",
                  "value": "1557911235"
                }
              ]
            }
          }
        ]
      }
    ]
  }
}```

**Comment by gsmachado** [comment]
@nimmortal great observation.
I'm preparing the 1.0.10 with the basic getapplicationlogs call. I will release that anyway.
Then, once the new structure for "notifcations" is added, I can make another release.

**Comment by gsmachado** [comment]
Just FYI: release [1.0.10](https://github.com/neow3j/neow3j/releases/tag/1.0.10) is out.

**Comment by gsmachado** [comment]
@nimmortal: did you have the chance to extend the notification format and include the map?

**Comment by nimmortal** [comment]
@gsmachado sorry for late response, I was busy on different project. I am going to implement this during NEO contract developing.

**Comment by claudemiller** [comment]
The `NeoApplicationLog` class has been adapted with pull-request #91. The `StackItem`, that is now used in the notification's state json object, supports the map data structure (among others).
In my opinion this issue is resolved.
@nimmortal  Do you agree?
The `StackItem` implementation is not yet available in a SNAPSHOT release but it is available in neow3j's `develop` branch.

**Comment by nimmortal** [comment]
@claudemiller, yes, I assume that this is solved.

**Comment by claudemiller** [comment]
Waiting for next neow3j release.

**Comment by claudemiller** [comment]
Resolved in release 2.3.0
